### DISCUSSION TITLE
How to view generated equations?

### DISCUSSION THREAD

**Discussion by jstr045329** [SpiceSharp/SpiceSharp#147] [General]
Hi, 

Is there a way to view the difference equations generated by SpiceSharp? 

Jim

**Comment by svenboulanger** [comment]
Hi @jstr045329 ,

All equations and derivatives are hard-coded in Spice#, so you can't view the equations directly. If you are interested in reverse-engineering them, then you can always refer to their behaviors which are all stored [here](https://github.com/SpiceSharp/SpiceSharp/tree/master/SpiceSharp/Components), or you can find online documentation as they're all pretty standard.

Yours sincerely,
Sven

**Reply by jstr045329** [reply]
But don't SPICE simulators normally use a netlist to generate a set of differential/difference equations in C, compile them, and then display the results? Does SpiceSharp operate the same way?

**Reply by svenboulanger** [reply]
I don't know of any Spice simulators that compile C-code (although maybe LTSpice had some option for compiling some small part that can optimize performance in some cases - but I doubt that it actually compiles C-code). It was certainly not a part of the "original" Spice 3f5 that I used as a reference.

All SPICE simulators that I know of use Modified Nodal Analysis (I wrote a tutorial about that [here](https://spicesharp.github.io/SpiceSharp/articles/custom_components/modified_nodal_analysis.html)). Each component/model will have hardcoded (compiled) equations that will ultimately "load" a big matrix with values that *represent* the KCL equations. You can view the Y-matrix and the right-hand-side vector if you want, but they won't show you the actual equations that generated those values.

I.e. without looking at the code, you can only observe the results of those equations, not the equations themselves.

**Reply by jstr045329** [reply]
@svenboulanger  thank you so much! It's been a minute since I took circuits. I forgot that you can solve a matrix.
